.github/workflows/detect_sha_changes.yml

```diff
@@ -28,9 +28,9 @@ jobs:
       - name: Fetch merge base SHA from API
         run: |
           my_merge_base_cmd="gh api repos/${{ github.repository }}/compare/${{ github.event.pull_request.base.sha }}...${{ github.event.pull_request.head.sha }} | jq -r '.merge_base_commit.sha'"
-          echo $my_merge_base_cmd
-          my_merge_base=$(eval $my_merge_base_cmd)
-          echo "MY_MERGE_BASE_SHA=$my_merge_base" >> $GITHUB_ENV
+          echo "${my_merge_base_cmd}"
+          my_merge_base="$(eval "${my_merge_base_cmd}")"
+          printf 'MY_MERGE_BASE_SHA=%s\n' "${my_merge_base}" >> "${GITHUB_ENV}"
```

Comment:
I'm curious what this switch to `printf` protects against; I wonder if it should be `%q` instead? Either way, this is certainly not less safe than what we had before, and we don't need this to be perfectly safe anyway.

Reply:
I'm responsible for this change and it was in response to a misinterpretation of the linter: the linter just pointed just to the `run` line, and I guessed what the issue was.  After trying various changes, I think that it would have been enough to simply sprinkle more double quotes around variables, but by then I already made this change.